### PR TITLE
[ENHANCEMENT] Add support for connection_string and url to DatabaseStoreBackend #2073

### PR DESCRIPTION
**Pull request by jcampbell**
[ENHANCEMENT] to support #2073

### CODE REVIEW

**Review by anthonyburdi — COMMENTED** [review]
The changes in this commit look mostly good to me (see single comment): https://github.com/great-expectations/great_expectations/pull/2088/commits/ee7344ee31dcd1aee4bb6bf803c93f77ae252791, I have not reviewed the rest

File: great_expectations/data_context/store/database_store_backend.py [hunk 1]
```diff
@@ -115,13 +117,7 @@ def _build_engine(self, credentials, **kwargs) -> "sa.engine.Engine":
...
         # Update credentials with anything passed during connection time
         drivername = credentials.pop("drivername")
-        schema_name = credentials.pop("schema_name", None)
-        if schema_name is not None:
-            logger.warning(
-                "schema_name specified creating a URL with schema is not supported. Set a default "
-                "schema on the user connecting to your database."
-            )
-
+        self._schema_name = credentials.pop("schema_name", None)
```
Comment (anthonyburdi) [inline great_expectations/data_context/store/database_store_backend.py]: Should this be above line 79 where it is first referenced?

**Review by eugmandel — APPROVED** [review]
Great!

File: great_expectations/data_context/store/database_store_backend.py (hunk 1 above)
Comment (Shinnnyshinshin) [reply great_expectations/data_context/store/database_store_backend.py]: You're absolutely right :) Thanks for catching this

File: great_expectations/data_context/store/database_store_backend.py [hunk 2]
```diff
@@ -55,12 +55,32 @@ def __init__(
...
+        elif connection_string is not None:
+            self.engine = sa.create_engine(connection_string, **kwargs)
+        elif url is not None:
+            self.drivername = urlparse(url).scheme
+            self.engine = sa.create_engine(url, **kwargs)
+        else:
+            raise ge_exceptions.InvalidConfigError(
+                "Credentials, url, connection_string, or an engine are required for a DatabaseStoreBackend."
+            )
+        # <WILL> Add an additional check here?
```
Comment (Shinnnyshinshin) [inline great_expectations/data_context/store/database_store_backend.py]:
```suggestion
```

File: great_expectations/data_context/store/database_store_backend.py [hunk 3]
```diff
@@ -55,12 +60,31 @@ def __init__(
...
+        elif url is not None:
+            self.drivername = urlparse(url).scheme
+            self.engine = sa.create_engine(url, **kwargs)
+        else:
+            raise ge_exceptions.InvalidConfigError(
+                "Credentials, url, connection_string, or an engine are required for a DatabaseStoreBackend."
+            )
 
-        meta = MetaData(schema=schema)
+        meta = MetaData(schema=self._schema_name)
```
Comment (Shinnnyshinshin) [inline great_expectations/data_context/store/database_store_backend.py:87]: `self._schema_name` can be assigned by the `_build_engine()` function, which has run by the time the code gets to this line

File: great_expectations/data_context/store/database_store_backend.py [hunk 4]
```diff
@@ -31,21 +34,23 @@
...
         key_columns,
         fixed_length_key=True,
-        suppress_store_backend_id=False,
-        manually_initialize_store_backend_id: str = "",
+        credentials=None,
+        url=None,
+        connection_string=None,
+        engine=None,
         store_name=None,
+        suppress_store_backend_id=False,
```
Comment (Shinnnyshinshin) [inline great_expectations/data_context/store/database_store_backend.py:45]: `store_name` and `supress_store_backend_id` were re-added to the signature due to tests failing.

**Pull request merged**